File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l, r = 0, len(s)-1
        while (l < r):
            # isalnum() : returns True if all the characters are alphanumeric,
            while l < r and not s[l].isalnum():
                l += 1
            while l < r and not s[r].isalnum():
                r -= 1
            if s[l].lower() != s[r].lower():
                return False
            l +=1 
            r -= 1
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.c
bool isPalindrome(char * s){
    int slen = 0;
    while (s[slen]!='\0'){
        slen++;
    }

    int l = 0, r = slen;
    while(l<r){
        while (l<r && !(isalnum(s[l]))){l++;}
        while (l<r && !(isalnum(s[r]))){r--;}
        if (tolower(s[l])!=tolower(s[r])){
            return false;
        }
        l++; r--;
    }
    return true;
}